src/Raven.Server/Rachis/TimeoutEvent.cs

```diff
@@ -117,27 +117,41 @@ public void DisableTimeout()
             }
         }
 
+        private int _inProgress;
+
         public void ExecuteTimeoutBehavior()
         {
-            lock (this)
-            {
-                if (Disable)
-                    return;
+            var action = _timeoutHappened;
```

Comment:
I don't like this.
We need to do this, but we have to capture the `_timouteHappened` value _inside_ the lock

Reply:
done